### PR TITLE
Add timeout option to Savon client

### PR DESCRIPTION
**Pull request by downgba**

### CODE REVIEW

File: spec/correios_sigep/configuration_spec.rb [hunk 1]
```diff
@@ -80,5 +80,18 @@ module CorreiosSigep
         expect(subject.user).to eq 'user'
       end
     end
+
+    describe '#timeout' do
+      it 'return nil when unset' do
```
Comment (duduribeiro) [inline spec/correios_sigep/configuration_spec.rb]: it `returns`

File: spec/correios_sigep/configuration_spec.rb [hunk 2]
```diff
@@ -80,5 +80,18 @@ module CorreiosSigep
...
+        expect(subject.timeout).to_not be
+      end
+    end
+
+    describe '#timeout=' do
+      it 'can set the timeout in configuration' do
```
Comment (duduribeiro) [inline spec/correios_sigep/configuration_spec.rb]: it '`sets`

File: spec/correios_sigep/configuration_spec.rb [hunk 3]
```diff
@@ -80,5 +80,18 @@ module CorreiosSigep
         expect(subject.user).to eq 'user'
       end
     end
+
+    describe '#timeout' do
+      it 'return nil when unset' do
+        expect(subject.timeout).to_not be
```
Comment (duduribeiro) [inline spec/correios_sigep/configuration_spec.rb]: @fpgentil What u think about change all `to_not be` to `not_to be`?
I think it is more english readable 😄

**Comment by duduribeiro** [comment]
@downgba this error is because `savon` requires `rack` on *any* version. And the bundler tries to fetch the last one (2.0.1) that is not compatible with ruby < 2.2.2.
Please, can u add a dependency on `correios_sigep.gemspec`?
Add rack in `1.6.4` version.

Thanks so much, 🍻

File: spec/correios_sigep/configuration_spec.rb [hunk 4]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
...
-      it 'can set the user in configuration' do
+      it 'sets the user in configuration' do
         subject.user = 'user'
         expect(subject.user).to eq 'user'
       end
     end
+
+    describe '#timeout' do
+      it 'returns nil when unset' do
+        expect(subject.timeout).not_to be
```
Comment (fpgentil) [inline spec/correios_sigep/configuration_spec.rb]: Se o `subject.timeout` é de fato nil, acho que faz mais sentido:

```
expect(subject.timeout).to be_nil
```

File: spec/correios_sigep/configuration_spec.rb [hunk 5]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
     subject { described_class.new }
 
     describe '#administrative_code' do
-      it 'return nil when unset' do
-        expect(subject.administrative_code).to_not be
```
Comment (duduribeiro) [inline spec/correios_sigep/configuration_spec.rb]: @downgba can u please change all `to_not` for `not_to`? :P 
cc @fpgentil 
thanks =D

File: spec/correios_sigep/configuration_spec.rb [hunk 6]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
...
         expect(subject.password).to eq 'password'
       end
     end
 
 
     describe '#service_code' do
-      it 'return nil when unset' do
-        expect(subject.service_code).to_not be
+      it 'returns nil when unset' do
+        expect(subject.service_code).not_to be
```
Comment (fpgentil) [inline spec/correios_sigep/configuration_spec.rb]: O mesmo pro comentário ali:

```
expect(subject.service_code).to be_nil
```

File: spec/correios_sigep/configuration_spec.rb [hunk 7]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
...
         subject.contract = 'contract'
         expect(subject.contract).to eq 'contract'
       end
     end
 
     describe '#password' do
-      it 'return nil when unset' do
-        expect(subject.password).to_not be
+      it 'returns nil when unset' do
+        expect(subject.password).not_to be
```
Comment (fpgentil) [inline spec/correios_sigep/configuration_spec.rb]:
```
...to be_nil
```

File: spec/correios_sigep/configuration_spec.rb [hunk 8]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
...
         subject.card = 'card'
         expect(subject.card).to eq 'card'
       end
     end
 
     describe '#contract' do
-      it 'return nil when unset' do
-        expect(subject.contract).to_not be
+      it 'returns nil when unset' do
+        expect(subject.contract).not_to be
```
Comment (fpgentil) [inline spec/correios_sigep/configuration_spec.rb]: Aqui também! 😄

File: spec/correios_sigep/configuration_spec.rb [hunk 9]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
...
         subject.administrative_code = '12345'
         expect(subject.administrative_code).to eq '12345'
       end
     end
 
     describe '#card' do
-      it 'return nil when unset' do
-        expect(subject.card).to_not be
+      it 'returns nil when unset' do
+        expect(subject.card).not_to be
```
Comment (fpgentil) [inline spec/correios_sigep/configuration_spec.rb]: ☝️ 👇

File: spec/correios_sigep/configuration_spec.rb [hunk 10]
```diff
@@ -3,82 +3,95 @@ module CorreiosSigep
     subject { described_class.new }
 
     describe '#administrative_code' do
-      it 'return nil when unset' do
-        expect(subject.administrative_code).to_not be
+      it 'returns nil when unset' do
+        expect(subject.administrative_code).not_to be
```
Comment (fpgentil) [inline spec/correios_sigep/configuration_spec.rb]: Último 😄

**Comment by duduribeiro** [comment]
🎆

File: lib/correios_sigep/logistic_reverse/base_client.rb [hunk 11]
```diff
@@ -1,8 +1,19 @@
 module CorreiosSigep
   module LogisticReverse
     class BaseClient
+
```
Comment (duduribeiro) [inline lib/correios_sigep/logistic_reverse/base_client.rb]: Only one thing.. can u please remove this extra line after `BaseClient`

**Comment by duduribeiro** [comment]
Can u please change 
`lib/correios_sigep/version.rb` 
from `VERSION = '0.4.0'` to `VERSION = '0.5.0'` ?

**Pull request merged**